src/main/java/com/chat/yourway/service/ChatMessageServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.chat.yourway.service;
+
+import com.chat.yourway.dto.request.ReceivedMessageDto;
+import com.chat.yourway.dto.response.MessageResponseDto;
+import com.chat.yourway.mapper.MessageMapper;
+import com.chat.yourway.service.interfaces.ChatMessageService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ChatMessageServiceImpl implements ChatMessageService {
+
+  private final SimpMessagingTemplate simpMessagingTemplate;
+  private final MessageMapper messageMapper;
+
+  @Value("${socket.dest-prefixes}")
+  private String[] destPrefixes;
```

Comment:
Optional - its better to autowire such variables at constructor. It will improve testing-ability of the code. because in case when u will need to add some value in tests - you will need to use reflection in order to fill this field